src/renderer/components/swap/Swap.tsx

```diff
@@ -51,12 +57,15 @@ type SwapProps = {
   poolDetails?: PoolDetails
   walletBalances?: O.Option<NonEmptyWalletBalances>
   nativeTxFee: NativeFeeRD
+  targetChainFee?: ChainFeeRD
+  sourceChainFee?: ChainFeeRD
```

Comment:
Provide a single property `fees` (similar to `AddDeposit`)

```
export type SwapFees = { source: BaseAmount; target: BaseAmount }
export type SwapFeesRD = RD.RemoteData<Error, SwapFees>
```
^ Similar to [`DepositFeesRD`](https://github.com/thorchain/asgardex-electron/blob/426654b801b689740ec326fc031a815cc9c10ed9/src/renderer/services/chain/types.ts#L34-L35)

With that ^, more logic can be moved from view into service layer.